### PR TITLE
Hughsie/fu device set target

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@superm1 the last commit is the first udev emulated device -- if this PR looks good (and works!) then I can dig out a whole ton of hardware and add a lot more CI tests. This should allow us to raise the coverage number significantly.

On a related question -- how would you feel about removing the umockdev tests like nvme_test.py and replacing them with a full blown emulated device update?

**Comment by superm1** [comment]
> On a related question -- how would you feel about removing the umockdev tests like nvme_test.py and replacing them with a full blown emulated device update?

Totally makes sense.

File: libfwupdplugin/fu-backend.c [hunk 1]
```diff
@@ -207,10 +208,11 @@ fu_backend_create_device(FuBackend *self, const gchar *backend_id, GError **erro
...
 
 	if (klass->create_device == NULL) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "not implemented");
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "->create_device is not implementedin %s",
```
Comment (superm1) [inline libfwupdplugin/fu-backend.c:214]: nit; missing space

**Review by superm1 — APPROVED** [review]
Just one nit; LGTM.

File: libfwupdplugin/fu-backend.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-backend.c:214]: Fixed, thanks!

**Pull request merged**